RELEASE.md

```diff
@@ -1,3 +1,8 @@
+# Release 0.17.7
```

Comment:
I noticed that there is no new section on the release notes for the upcoming release, and had temporarily named the upcoming release as `0.17.7` for now - feel free to suggest any new version otherwise